packages/flutter/lib/src/widgets/route_notification_messages.dart

```diff
@@ -14,28 +15,35 @@ class RouteNotificationMessages {
   // ignore: unused_element
   RouteNotificationMessages._();
 
-  /// When the engine is Web notify the platform for a route change.
-  static void maybeNotifyRouteChange(String routeName, String previousRouteName) {
+  /// Notifies the platform for a route information change.
+  ///
+  /// See also:
+  ///
+  ///  * [SystemNavigator.routeInformationUpdated], which handles subsequent
+  ///    navigation requests.
+  static void notifyRouteInformationChange(RouteInformation routeInformation) {
     if(kIsWeb) {
```

Comment:
@goderbauer 
we have to do this check because method channel invoke method will throw if there is no corresponding listener on the engine side.

Reply:
Should the navigator channel be an OptionalMethodChannel to avoid that problem?